builtins/amp-img.js

```diff
@@ -141,6 +145,17 @@ export class AmpImg extends BaseElement {
     }
 
     this.propagateAttributes(ATTRIBUTES_TO_PROPAGATE, this.img_);
+
+    if (this.useNativeSrcset_) {
+      this.propagateAttributes(['src', 'srcset', 'sizes'], this.img_);
+
+      const nativeSizes = this.element.getAttribute('native-sizes');
+      if (nativeSizes) {
+        // Otherwise sizes will be propagated normally if defined
+        this.img_.setAttribute('sizes', nativeSizes);
```

Comment:
I'm slightly puzzled by this code, but there might be a misunderstanding on the goal. When 'amp-img-native-srcset' is set we should never even initialize `this.srcset_`. And we talked about generating a sizes attribute if `native-sizes` is not provided, so the browser can choose the right one from the `srcset` based on screen size only (emulating the behavior of our current implementation).

Reply:
I preserved this code because of the edge case where `srcset` is not supported (IE, Opera Mini) but the user provided `srcset` instead of `src`. But I think you are right, we should not initialize `srcset` at all, and it probably suffices to just persist the first url in `srcset` like we currently do in the preconnect callback. 

Automatically generating `sizes` is an outstanding FR as part of this issue, just not addressed in this PR.

Reply:
I'm not sure supporting browsers without srcset support is worth it anymore since the fallback to just src is fine.

If you wanted to do that, then we should only call the code if `'srcset' in img` is false, but not do the work and then override it.